modules/reactor/doc/content/source/meshgenerators/PolygonConcentricCircleMeshGenerator.md

```diff
@@ -0,0 +1,80 @@
+- `polygon_size`: the size of the polygon enclosure in terms of either `apothem` or `radius` (selected by another parameter, `polygon_size_style`)
+
+!media media/meshgenerators/polygon.png
+      style=display: block;margin-left:auto;margin-right:auto;width:40%;
+      id=schematic
+      caption=A schematic drawing showing the different regions that can be generated by this `PolygonConcentricCircleMeshGenerator` object.
+
+If concentric circle regions are needed, `ring_radii` must be provided to define the outer radius of each ring. If duct regions are needed, `duct_sizes` must be provided to define the inner boundary of each duct. Similar to `polygon_size`, `duct_sizes` can either be apothems or radii of the corresponding ducts, decided by `duct_sizes_style`.
+
+## Mesh Information
+
+The parameters that are needed for meshing can be categorized into azimuthal and radial information, which determine the mesh density.
+
+The azimuthal mesh discretization is controlled by `num_sectors_per_side`, which is a vector that has the same length as `num_sides`. Each element of `num_sectors_per_side` must be an even integer, which is the number of azimuthal intervals of the corresponding side. The sides are indexed counterclockwise with the upper-right polygon side being the first side as shown in [Figure 2](#schematic).
+
+The radial mesh discretization is controlled by separate parameters for `rings`, `background` and `ducts` regions, which are `ring_intervals`, `background_intervals`, and `duct_intervals`, respectively.
+
+## MeshMeshData Generation
+
+When `num_sides` is set as 6, the hexagonal meshes produced by `PolygonConcentricCircleMeshGenerator` can be used by [`PatternedHexMeshGenerator`](/PatternedHexMeshGenerator.md) to create patterned meshes. This functionality relies on a series of `MeshMetaData` generated by `PolygonConcentricCircleMeshGenerator`, including:
+
+- `pitch_meta`: a `Real` type `MeshMetaData` that provides the side-to-side distance of the hexagon generated.
+- `pattern_pitch_meta`: a `Real` type `MeshMetaData` that is similar to `pitch_meta`. This is only used when the mesh produced here is stitched with assembly hexagon meshes generated by `PatternedHexMeshGenerator`.
+- `background_intervals_meta`: a `MeshMetaData` that makes the value of the input parameter `background_intervals` available to other mesh generators.
+- `node_id_background_meta`: a `MeshMetaData` that records the maximum node id of the background nodes to provide a reference for node modifications.
+- `max_radius_meta`: a `Real` type `MeshMetaData` that provides the maximum radius of the ring regions so that `PatternedHexMeshGenerator` does not compromise the circular geometry.
+- `num_sectors_per_side_meta`: a `MeshMetaData` that makes the value of the input parameter `num_sectors_per_side` available to other mesh generators.
+- `azimuthal_angle_meta`: a vector `MeshMetaData` that contains sorted azimuthal angles of all the nodes on the external boundary.
```

Comment:
```suggestion
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/pitch_meta): a `Real` type `MeshMetaData` that provides the side-to-side distance of the hexagon generated.
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/pattern_pitch_meta): a `Real` type `MeshMetaData` that is similar to [!param](/Mesh/PolygonConcentricCircleMeshGenerator/pitch_meta). This is only used when the mesh produced here is stitched with assembly hexagon meshes generated by `PatternedHexMeshGenerator`.
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/background_intervals_meta): a `MeshMetaData` that makes the value of the input parameter `background_intervals` available to other mesh generators.
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/node_id_background_meta): a `MeshMetaData` that records the maximum node id of the background nodes to provide a reference for node modifications.
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/max_radius_meta): a `Real` type `MeshMetaData` that provides the maximum radius of the ring regions so that `PatternedHexMeshGenerator` does not compromise the circular geometry.
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/num_sectors_per_side_meta): a `MeshMetaData` that makes the value of the input parameter [!param](/Mesh/PolygonConcentricCircleMeshGenerator/num_sectors_per_side) available to other mesh generators.
- [!param](/Mesh/PolygonConcentricCircleMeshGenerator/azimuthal_angle_meta): a vector `MeshMetaData` that contains sorted azimuthal angles of all the nodes on the external boundary.
```